RELEASE/scripts/sl_ascend.ash

```diff
@@ -13847,6 +13898,10 @@ boolean sl_tavern()
 	}
 	print("In the tavern! Layout: " + tavern, "blue");
 	boolean [int] locations = $ints[3, 2, 1, 0, 5, 10, 15, 20, 16, 21];
+
+	// Infrequent compunding issue, reset maximizer
+	resetMaximize();
```

Comment:
Can you clarify what's going on here? "Infrequent compounding" worries me. Randomly resetting the maximizer also worries me.

Reply:
I was having some weird performance in other areas and added a debug to print the Current Maximizer string. I would randomly see things like this:

Boondoggle_The_Magnificent_20190724.txt:> Current Maximize String: -equip makeshift garbage shirt,-equip broken champagne bottle,-equip Snow Suit,200cold damage 20max,200hot damage 20max,200spooky damage 20max,20
0stench damage 20max,100ml 149max,200cold damage 20max,200hot damage 20max,200spooky damage 20max,200stench damage 20max,100ml 149max,200cold damage 20max,200hot damage 20max,200spooky damage 20max,200stench dama
ge 20max,100ml 149max,200cold damage 20max,200hot damage 20max,200spooky damage 20max,200stench damage 20max,100ml 149max,200cold damage 20max,200hot damage 20max,200spooky damage 20max,200stench damage 20max,100
ml 149max,200cold damage 20max,200hot damage 20max,200spooky damage 20max,200stench damage 20max,100ml 149max,200cold damage 20max,200hot damage 20max,200spooky damage 20max,200stench damage 20max,100ml 149max

Reply:
Oh, you're totally right. I didn't realize that we handled the tavern in a super weird way (we call `sl_tavern()` in a loop from `L3_tavern) and so the maximizer isn't reset like it should be. You're doing the right thing. Your condition on entering the tavern loop is, I think, also correct - I was worried about getting preeempted by a higher-priority task while we're doing tavern, but that's impossible.